### PR TITLE
Use centralized checks selection for both hosts and clusters

### PR DESCRIPTION
**Pull request by nelsonkopliku**
# Description

This PR:
- adds the cluster checks selection saga in the centralized checks selection saga
- makes the new centralized checks selection used both in host and cluster checks selection pages.
- cleans up legacy

Note that the changes in the legacy `assets/js/components/ClusterDetails/ChecksSelection.jsx` and test are basically transient as that component will be removed soon in favor of the generalized one in `assets/js/components/ChecksSelection/ChecksSelection.jsx`

## How was this tested?

Automated tests.

### CODE REVIEW

File: assets/js/components/ClusterDetails/ChecksSelection.jsx [hunk 1]
```diff
@@ -40,9 +46,12 @@ const getGroupSelectedState = (checks, selectedChecks) => {
...
 
-  const { saving, savingError, savingSuccess } = useSelector(
-    (state) => state.clusterChecksSelection
-  );
+  const { status } = useSelector(getClusterCheckSelection(clusterId));
+  const { saving, savingError, savingSuccess } = {
+    saving: isSaving(status),
+    savingSuccess: isSuccessfullySaved(status),
+    savingError: isSavingFailed(status),
+  };
```
Comment (dottorblaster) [inline assets/js/components/ClusterDetails/ChecksSelection.jsx:54]:
```suggestion
const saving = isSaving(status);
const savingSuccess = isSuccessfullySaved(status);
const savingError = isSavingFailed(status);
```

File: assets/js/state/checksSelection.js [hunk 2]
```diff
@@ -5,6 +5,10 @@ const SAVING = 'SAVING';
 const SUCCESSFULLY_SAVED = 'SUCCESSFULLY_SAVED';
 const SAVING_FAILED = 'SAVING_FAILED';
 
+export const isSaving = (status) => SAVING === status;
+export const isSuccessfullySaved = (status) => SUCCESSFULLY_SAVED === status;
+export const isSavingFailed = (status) => SAVING_FAILED === status;
+
```
Comment (dottorblaster) [inline assets/js/state/checksSelection.js:11]: The idiomatic way to do this is creating selectors that return boolean values. `reselect` would help, but we still didn't introduce that.

**Review by dottorblaster — CHANGES_REQUESTED** [review]
A couple changes and then we can merge

File: assets/js/state/checksSelection.js (hunk 2 above)
Comment (nelsonkopliku) [reply assets/js/state/checksSelection.js:11]: Cool! Would you mind tracking that so we can plan for a refactor?

Comment (dottorblaster) [reply assets/js/state/checksSelection.js:11]: Yes absolutely! Tracking that on Jira right now, could you still move this to selectors and base the functions on `resourceID`?

Comment (nelsonkopliku) [reply assets/js/state/checksSelection.js:11]: sure. You meant `targetID` I guess :smile:

Comment (dottorblaster) [reply assets/js/state/checksSelection.js:11]: `whateverID`, I just get lost into these names :joy:

**Pull request merged**